doc/tutorials/using_cat_on_up2.rst

```diff
@@ -0,0 +1,164 @@
+for the HV shell (refer to :ref:`getting-started-up2` for setup instructions).
+
+#. Check CAT ability with cupid, first run ``cpuid 0x10 0x0``, the return value ebx[bit 2]
+   report the L2 CAT is supported. Then run ``cpuid 0x10 0x2`` to query L2 CAT capability,
+   the return eax[bit 4:0] report cache mask has 8 bit, edx[bit 15:0] report 4 CLOS are
+   supported
+
+   .. code-block:: none
+
+      ACRN:\>cpuid 0x10 0x0
+      cpuid leaf: 0x10, subleaf: 0x0, 0x0:0x4:0x0:0x0
+
+      ACRN:\>cpuid 0x10 0x2
+      cpuid leaf: 0x10, subleaf: 0x2, 0x7:0x0:0x0:0x3
+
+#. Check PCPU IDs of each VM, the ``vcpu_list`` shows that VM0 is running on PCPU0,
+   and VM1 is running on PCPU1
+
+   .. code-block:: none
+
+      ACRN:\>vcpu_list
+
+      VM ID    PCPU ID    VCPU ID    VCPU ROLE    VCPU STATE
+      =====    =======    =======    =========    ==========
+        0         0          0        PRIMARY       Running
+        1         1          0        PRIMARY       Running
+
+#. Set CLOS with ``wrmsr <reg_num> <value>``, we want VM1 to use the lower 6 ways of cache,
+   so CLOS0 is set to 0xc0 for the upper 2 ways, and CLOS1 is set to 0x3f for the lower
+   6 ways.
+
+   .. code-block:: none
+
+      ACRN:\>wrmsr  -p1 0xd10  0xc0
+      ACRN:\>wrmsr  -p1 0xd11  0x3f
+
+#. Attach COS1 to PCPU1, because MSR IA32_PQR_ASSOC [bit 63:32], so write 0x100000000
+   to it to use CLOS1
+
+   .. code-block:: none
+
+      ACRN:\>wrmsr   –p1   0xc8f    0x100000000
+
+Configure CAT for VM with VM Configuration
+##########################################
+
+#. CAT on ACRN can be enabled and configured by modifying source code, so the first
+   step is to clone the ACRN source code (if you haven't already):
+
+   .. code-block:: none
+
+      $ git clone https://github.com/projectacrn/acrn-hypervisor.git
+      $ cd acrn-hypervisor
+
+#. The predefined cache masks can be found at
+   ``hypervisor/arch/x86/configs/$(CONFIG_BOARD)/board.c``, for UP2 board, that is
+   ``hypervisor/arch/x86/configs/apl-up2/board.c``, you can change the mask values,
+   but note that the CLOS mask must have continuous bits, or a #GP fault can be triggered.
+
+   .. code-block:: none
+      :emphasize-lines: 3,7,11,15
+
+      struct platform_clos_info platform_clos_array[4] = {
+              {
+                      .clos_mask = 0xff,
+                      .msr_index = MSR_IA32_L2_MASK_0,
+              },
+              {
+                      .clos_mask = 0xff,
+                      .msr_index = MSR_IA32_L2_MASK_1,
+              },
+              {
+                      .clos_mask = 0xff,
+                      .msr_index = MSR_IA32_L2_MASK_2,
+              },
+              {
+                      .clos_mask = 0xff,
+                      .msr_index = MSR_IA32_L2_MASK_3,
+              },
+      };
+
+#. Set up CLOS at the VM config. If you want a VM to use one of the CLOSs, you need to find its
```

Comment:
```suggestion
#. Set up CLOS in the VM config. If you want a VM to use one of the CLOSs, you need to find its
```

Reply:
OK

Reply:
Hi, Kinder. I think we can merge the Doc this time，and  submit the Kconfig version in the future. Because the Kconfig fix relays on VM configuration code to be established. And the Doc can be helpful for people to debug CAT  features on ACRN.